sdks/python/apache_beam/transforms/ptransform.py

```diff
@@ -422,6 +423,35 @@ def with_output_types(self, type_hint):
     validate_composite_type_param(type_hint, 'Type hints for a PTransform')
     return super(PTransform, self).with_output_types(type_hint)
 
+  def with_resource_hints(self, **kwargs):  # type: (...) -> PTransform
+    """Adds resource hints to the :class:`PTransform`.
+
+    Resource hints allow users to express constraints on the environment where
+    the transform should be executed.  Interpretation of the resource hints is
+    defined by Beam Runners. Runners may ignore unsupported the hints.
+
+    Args:
+      **kwargs: key-value pairs describing hints and their values.
+
+    Raises:
+      ValueError: if provided hints are unknown to the SDK. See
+        :mod:~apache_beam.transforms.resources` for a list of known hints.
+
+    Returns:
+      PTransform: A reference to the instance of this particular
+      :class:`PTransform` object.
+    """
+    self._resource_hints = resources.parse_resource_hints(kwargs)
```

Comment:
This should probably be additive, eg. 

transform.with_resource_hints(foo=x).with_resource_hints(bar=y)

should have both x and y.

Reply:
Good point, thanks. Fixed.